pgx/src/datum/sql_entity_graph/mod.rs

```diff
@@ -64,6 +64,72 @@ pub trait ToSql {
     fn to_sql(&self, context: &PgxSql) -> eyre::Result<String>;
 }
 
+/// Represents a pointer to a function that can transform a SqlGraphEntity to a SQL string
+///
+/// This is used to provide a facility for overriding the default SQL generator behavior using
+/// the `#[to_sql(path::to::function)]` attribute in circumstances where the default behavior is
+/// not desirable.
+///
+/// Implementations can invoke `ToSql::to_sql(entity, context)` on the unwrapped SqlGraphEntity
+/// type should they wish to delegate to the default behavior for any reason.
+pub type ToSqlFn = fn(&SqlGraphEntity, &PgxSql) -> eyre::Result<String>;
+
+/// Represents configuration options for tuning the SQL generator.
+///
+/// When an item that can be rendered to SQL has these options at hand, they should be
+/// respected. If an item does not have them, then it is not expected that the SQL generation
+/// for those items can be modified.
+///
+/// The default configuration has `enabled` set to `true`, and `callback` to `None`, which indicates
+/// that the default SQL generation behavior will be used. These are intended to be mutually exclusive
+/// options, so `callback` should only be set if generation is enabled.
+///
+/// When `enabled` is false, no SQL is generated for the item being configured.
+///
+/// When `callback` has a value, the corresponding `ToSql` implementation should invoke the
+/// callback instead of performing their default behavior.
+#[derive(Clone)]
+pub struct ToSqlConfig {
```

Comment:
```suggestion
pub struct ToSqlConfigEntity {
```

Otherwise we see duplicated docs:
![image](https://user-images.githubusercontent.com/130903/151850568-64dcdd52-17fd-4ba1-83f2-5b247ef2c7b9.png)